pkg/features/kube_features.go

```diff
@@ -375,9 +375,15 @@ const (
 	// owner: @davidz627
 	// alpha: v1.14
 	//
-	// Enables the in-tree storage to CSI Plugin migration feature.
+	// Enables the in-tree storage to CSI Plugin migration feature (Phase 1).
 	CSIMigration featuregate.Feature = "CSIMigration"
 
+	// owner: @ddebroy
+	// alpha: v1.17
+	//
+	// Disables in-tree storage plugins for CSI migration (Phase 2).
+	CSIMigrationInTreeOff featuregate.Feature = "CSIMigrationInTreeOff"
```

Comment:
Also, does this assume that all migrating volume types will complete at the same time? it seems likely that some will lag (either absolutely in implementation time, or specifically in a given cluster).

Per-volume "migration complete" seem more likely to be what we want (CSIMigrationGCE / CSIMigrationGCEComplete, etc). The other benefit of that is that once migration of a specific volume type goes GA, its migration gates can be locked on, then removed the next release, and we'll only need to keep around the migration gates for volume types that haven't completed migration yet.

Reply:
Concerns about the naming of the feature flag was indeed something that @davidz627 brought up a while back but I did not get around to changing it. I can change it to `CSIMigrationComplete` (or the previously suggested `CSIMigrationCommited`).

I can add support for per-plugin migration completion feature flags in the CSI Migrated Plugin Manager but was hesitant due to increasing the number of feature flags we need to track by 2x. Per-plugin migration completion flags will be necessary if we have scenarios where operators simultaneously use multiple plugins all of which may be in different states of migration. That is probably remote with the current initial set of cloud provider in-tree plugins being migrated but maybe there are hybrid cloud scenarios with stretched clusters where this may arise in the future (or say EBS and iSCSI if iSCSI is migrated out in the future).

Reply:
Added separate `MigrationComplete` feature flags for GCE PD and AWS EBS to start off with and removed `CSIMigrationInTreeOff`.